pediatric_rootlets/pediatric_rootlets.sh

```diff
@@ -0,0 +1,148 @@
+segment_if_does_not_exist(){
+  FILESEG="${SUBJECT}_${FILE_TYPE}_T2w_label-SC_mask.nii.gz"
+  FILESEGMANUAL="${PATH_DATA}/derivatives/labels/${SUBJECT}/anat/${FILESEG}"
+  echo
+  echo "Looking for manual segmentation: $FILESEGMANUAL"
+  if [[ -e $FILESEGMANUAL ]]; then
+    echo "Found! Using manual segmentation."
+    rsync -avzh $FILESEGMANUAL ${FILESEG}
+    sct_qc -i ${file_t2} -s ${FILESEG} -p sct_deepseg_sc -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  else
+    echo "Not found. Proceeding with automatic segmentation."
+    # Segment spinal cord
+    sct_deepseg_sc -i ${file_t2} -c t2 -qc ${PATH_QC} -qc-subject ${SUBJECT} -o ${FILESEG}
+  fi
+}
+
+# Detect PMJ if it does not exist
+detect_pmj_if_does_not_exist(){
+  FILEPMJ="${SUBJECT}_${FILE_TYPE}_T2w_label-PMJ_dlabel.nii.gz"
+  FILEPMJMANUAL="${PATH_DATA}/derivatives/labels/${SUBJECT}/anat/${FILEPMJ}"
+  echo
+  echo "Looking for manual PMJ detection: $FILEPMJMANUAL"
+  if [[ -e $FILEPMJMANUAL ]]; then
+    echo "Found! Using manual PMJ detection."
+    rsync -avzh $FILEPMJMANUAL ${FILEPMJ}
+    sct_qc -i ${file_t2} -s ${SUBJECT}_${FILE_TYPE}_T2w_label-SC_mask.nii.gz -d ${FILEPMJ} -p sct_detect_pmj -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  else
+    echo "Not found. Proceeding with automatic PMJ detection."
+    # Detect PMJ
+    sct_detect_pmj -i ${file_t2} -s ${SUBJECT}_${FILE_TYPE}_T2w_label-SC_mask.nii.gz -c t2 -o ${FILEPMJ} -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  fi
+}
+
+# Label vertebral levels if it does not exist
+label_if_does_not_exist(){
+  # Update global variable with segmentation file name
+  FILELABEL="${SUBJECT}_${FILE_TYPE}_T2w_labels-disc.nii.gz"
+  FILELABELMANUAL="${PATH_DATA}/derivatives/labels/${SUBJECT}/anat/${FILELABEL}"
+  echo "Looking for manual label: $FILELABELMANUAL"
+  if [[ -e $FILELABELMANUAL ]]; then
+    echo "Found! Using manual labels."
+    rsync -avzh $FILELABELMANUAL ${FILELABEL}
+    # Generate labeled segmentation from manual disc labels
+    sct_label_vertebrae -i ${file_t2} -s ${SUBJECT}_${FILE_TYPE}_T2w_label-SC_mask.nii.gz -discfile ${FILELABEL} -c t2 -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  else
+    echo "Not found. Proceeding with automatic labeling."
+    # Generate labeled segmentation
+    sct_label_vertebrae -i ${file_t2} -s ${SUBJECT}_${FILE_TYPE}_T2w_label-SC_mask.nii.gz -c t2 -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  fi
+}
+
+# SCRIPT STARTS HERE
+# ==============================================================================
+# Display useful info for the log, such as SCT version, RAM and CPU cores available
+sct_check_dependencies -short
+
+# Go to folder where data will be copied and processed
+cd $PATH_DATA_PROCESSED
+
+# Copy source images
+rsync -Ravzh ${PATH_DATA}/./${SUBJECT}/anat/${SUBJECT//[\/]/_}_*T2w.* .
+
+# Go to anat folder where all structural data are located
+cd ${SUBJECT}/anat
+
+# Define the names of the T2w files
+file_t2_composed=${SUBJECT}_rec-composed_T2w.nii.gz
+file_t2_top=${SUBJECT}_acq-top_run-1_T2w.nii.gz
```

Comment:
I think it would be more convenient not to use `.nii.gz` here. If you omit `.nii.gz`, then you can directly reuse the variable even for labels and segmentations, e.g., `${file_t2}_label-SC_mask.nii.gz` without a need to use the auxiliary variable `${FILE_TYPE}`

```suggestion
file_t2_composed=${SUBJECT}_rec-composed_T2w
file_t2_top=${SUBJECT}_acq-top_run-1_T2w
```

Reply:
Resolved in [8be3c0a](https://github.com/ivadomed/model-spinal-rootlets/pull/53/commits/8be3c0a6f86331f6f29ac1d38eff028090046f7d).